src/DSharp.Compiler/Compiler/ExpressionBuilder.cs

```diff
@@ -1235,11 +1291,24 @@ private Expression ProcessOpenParenExpressionNode(BinaryExpressionNode node)
                 isDelegateInvoke = true;
             }
 
+            if (leftExpression is MethodExpression methodExpression && methodExpression.IsExtensionMethod)
```

Comment:
this shouldn't be needed if the method is marked as static

Reply:
Not sure I follow what you mean here?

Reply:
Without the check if this is an extension method we explode because we don't have any the of the method parameters used. It doesn't seem to want to work in the usual static path :(